File: php/websocket_server.php
<?php
require __DIR__ . '/vendor/autoload.php';
use Ratchet\Server\IoServer;
use Ratchet\Http\HttpServer;
use Ratchet\WebSocket\WsServer;
use Ratchet\ConnectionInterface;
use Ratchet\MessageComponentInterface;

class EchoServer implements MessageComponentInterface {
    public function onOpen(ConnectionInterface $conn) {
        echo "A client just connected\n";
    }

    public function onMessage(ConnectionInterface $from, $msg) {
        echo sprintf("Received message from client: %s\n", $msg);
        $from->send(sprintf("Pong: %s", $msg));
    }

    public function onClose(ConnectionInterface $conn) {
        echo "A client just disconnected\n";
    }

    public function onError(ConnectionInterface $conn, \Exception $e) {
        echo sprintf("An error has occurred: %s\n", $e->getMessage());
        $conn->close();
    }
}

$server = IoServer::factory(
    new HttpServer(
        new WsServer(
            new EchoServer()
        )
    ),
    8009
);

$server->run();

File: ruby/websocket_server.rb
require 'em-websocket'
require 'logger'

PORT = 8009
logger = Logger.new(STDOUT)

EventMachine.run do
  EventMachine::WebSocket.start(host: 'localhost', port: PORT) do |ws|
    ws.onopen do
      logger.info("Client connected")
    end

    ws.onmessage do |msg|
      logger.info("Received message: #{msg}")
      ws.send "Pong: #{msg}"
    end

    ws.onclose do
      logger.info("Client disconnected")
    end

    ws.onerror do |error|
      logger.error("Error occurred: #{error.message}")
    end
  end

  logger.info("WebSocket server started on port #{PORT}")
end
